dms/controllers/portal.py

```diff
@@ -1,32 +1,24 @@
-# Copyright 2020 Tecnativa - Víctor Martínez
+# Copyright 2020-2021 Tecnativa - Víctor Martínez
 import base64
-import logging
 
 from odoo import _, http
-from odoo.exceptions import AccessError, MissingError
 from odoo.http import request
 from odoo.osv.expression import OR
 
 from odoo.addons.portal.controllers.portal import CustomerPortal
 from odoo.addons.web.controllers.main import content_disposition, ensure_db
 
-_logger = logging.getLogger(__name__)
-
 
 class CustomerPortal(CustomerPortal):
     def _dms_check_access(self, model, res_id, access_token=None):
-        try:
-            item = request.env[model].browse(res_id)
-        except (AccessError, MissingError):
-            return False
-
         if access_token:
+            item = request.env[model].sudo().browse(res_id)
             if not item.check_access_token(access_token):
                 return False
         else:
+            item = request.env[model].browse(res_id)
             if not item.sudo(request.env.user.id).check_access("read", False):
                 return False
-
         return item
```

Comment:
Why https://github.com/odoo/odoo/blob/8703ff1e3d9be6f2f5fce2e8c4e62589b05133fb/addons/portal/controllers/portal.py#L233-L244 is not enough?

Reply:
`_document_check_access` function is not enough for some reasons:
- Odoo allways return `document_sudo` and it's totally incorrect (in dms, if we use inherit permissions is important NOT apply `.sudo()` or we will show more records of which you should, all sales for example).
- `access_token` is checked allways with model but in dms is necessary to check with directory (if we try to access to `dms.file` model) or check parent directories (if we try to access to subdirectories).

Reply:
> Odoo allways return `document_sudo` and it's totally incorrect (in dms, if we use inherit permissions is important NOT apply `.sudo()` or we will show more records of which you should, all sales for example).

but then, what happens if you share a tokenized URL of a directory that inherits permissions? 🤔 Is that a way to surpass permissions?



> `access_token` is checked allways with model but in dms is necessary to check with directory (if we try to access to `dms.file` model) or check parent directories (if we try to access to subdirectories).

OK that makes sense.